### DISCUSSION TITLE
OPDS Filename different in KOreader?

### DISCUSSION THREAD

**Discussion by colinkarr** [koreader/koreader#10039] [Q&A]
I've been playing around with the different ways to download books to KOreader from Calibre. The most appealing to me is via the OPDS connection out of KOreader (Magnifying Glass > OPDS Catalog > Local calibre library). However, I've noticed that the filename is different ({author name} - {title}) from the filename that is generated when I Send to Device straight out of Calibre, or when I access Calibre's OPDS content server from the browser ({title} - {author}). This causes sorting and syncing issues with my other devices. I know I can change the filename at the moment of download, but is there anyway to customize the filename template in KOreader?

**Comment by NiLuJe** [comment]
Not... really. Unless there's an OPDS hat-trick with an extension to the standard that would offer a ready-to-use filename we could use?

Right now we build it based on the title of the item, prepended by the author, if any.

https://github.com/koreader/koreader/blob/198a84a772b6bc2f2b652c7537d41a61405efca4/plugins/opds.koplugin/opdsbrowser.lua#L579-L582

**Reply by Frenzie** [reply]
@NiLuJe Depending on the server a filename might be provided through HTTP headers. The problem is those are often something like a meaningless ID number.

**Comment by colinkarr** [comment]
Understood - thank you! Not the end of the word, but consistency in
filenames would be great. Thanks for your quick response!

On Wed, Jan 18, 2023 at 11:33 AM NiLuJe ***@***.***> wrote:

> Not... really. Unless there's an OPDS hat-trick with an extension to the
> standard that would offer a ready-to-use filename we could use?
>
> Right now we build it based on the title of the item, prepended by the
> author is there is any.
>
>
> https://github.com/koreader/koreader/blob/198a84a772b6bc2f2b652c7537d41a61405efca4/plugins/opds.koplugin/opdsbrowser.lua#L579-L582
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/koreader/koreader/discussions/10039#discussioncomment-4719894>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHTMYOBRKOFBDAFNL7AL3ZTWTASMXANCNFSM6AAAAAAT7JD4GU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Under the Mercy,
Colin Karr
***@***.***
913.961.6299

**Comment by hius07** [comment]
Comment by the calibre creator:
"filenames when downloading from a server are ultimately controlled by the client. The calibre server does set the Content-disposition http header with a suggested filename (which you can control via plugboards) but clients are free to ignore that."
